bluemira/builders/cryostat.py

```diff
@@ -0,0 +1,200 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+Cryostat builder
+"""
+from typing import Dict, List, Tuple, Type, Union
+
+import numpy as np
+
+from bluemira.base.builder import Builder
+from bluemira.base.components import Component, PhysicalComponent
+from bluemira.base.designer import Designer
+from bluemira.base.parameter_frame import NewParameter as Parameter
+from bluemira.base.parameter_frame import NewParameterFrame as ParameterFrame
+from bluemira.base.parameter_frame import parameter_frame
+from bluemira.builders.tools import circular_pattern_component
+from bluemira.display.palettes import BLUE_PALETTE
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.tools import make_circle, make_polygon, revolve_shape
+
+
+class Cryostat:
+    """
+    Cryostat Component Manager TODO
+    """
+
+    def __init__(self, component: Component):
+        super().__init__()
+        self._component = component
+
+    def component(self) -> Component:
+        """
+        Return component
+        """
+        return self._component
+
+
+@parameter_frame
+class CryostatDesignerParams:
+    """
+    Cryostat designer parameters
+    """
+
+    g_cr_ts: Parameter[float]
+
+
+@parameter_frame
+class CryostatBuilderParams:
+    """
+    Cryostat builder parameters
+    """
+
+    g_cr_ts: Parameter[float]
+    n_TF: Parameter[int]
+    tk_cr_vv: Parameter[float]
+    # TODO add to Parameter default = 5 chickens
+    well_depth: Parameter[float]
+    x_g_support: Parameter[float]
+    # TODO add to Parameter default = 2
+    x_gs_kink_diff: Parameter[float]
+    # TODO add to Parameter default (z gravity support) = -15 chickens
+    z_gs: Parameter[float]
+
+
+class CryostatDesigner(Designer[BluemiraFace]):
+    """
+    Designer for the cryostat
+    """
+
+    param_cls: Type[CryostatDesignerParams] = CryostatDesignerParams
+
+    def __init__(
+        self,
+        params: Union[ParameterFrame, Dict],
+        cryo_ts_xz: BluemiraFace,
+    ):
+        super().__init__(params)
+        self.cryo_ts_xz = cryo_ts_xz
+
+    def run(self) -> Tuple[float]:
+        """
+        Cryostat designer run method
+        """
+        bound_box = self.cryo_ts_xz.bounding_box
+        z_max = bound_box.z_max
+        x_max = bound_box.x_max
+        x_out = x_max + self.params.g_cr_ts.value
+        z_top = z_max + self.params.g_cr_ts.value
+        return x_out, z_top
+
+
+class CryostatBuilder(Builder):
+    """
+    Builder for the cryostat
+    """
+
+    CRYO = "Cryostat VV"
+    param_cls: Type[CryostatBuilderParams] = CryostatBuilderParams
+
+    def build(self) -> Cryostat:
+        """
+        Build the cryostat component.
+        """
+        self.x_out, self.z_top = self.designer.run()
```

Comment:
In the plasma builder, I've made these local variables and passed them into the `build_*` methods. That way, it's not possible to call, e.g., `build_xz` before `self.x_out` is defined.